src/thread.c

```diff
@@ -10,7 +10,18 @@
 
 #include "rwlock.h"
 #include "tlocal.h"
+#include "error.h"
 
+// Thread
+
+MTY_Error MTY_ThreadGetError(MTY_Thread *ctx)
+{
+	// `ctx` is not used since errors are managed via tlocal state
```

Comment:
if the input argument is of no consequence, then let's remove it. the API you designed appears to always return the next error of the current thread, regardless of which thread the user asked for. in that case, there's no need for a data structure (MTY_Queue) whose purpose is to mediate access among many producer threads and one consumer thread.